python2/koans/about_classes.py

```diff
@@ -26,24 +26,24 @@ def set_name(self, a_name):
 
     def test_init_method_is_the_constructor(self):
         dog = self.Dog2()
-        self.assertEqual(__, dog._name)
+        self.assertEqual('Paul', dog._name)
 
     def test_private_attributes_are_not_really_private(self):
         dog = self.Dog2()
         dog.set_name("Fido")
-        self.assertEqual(__, dog._name)
+        self.assertEqual("Fido", dog._name)
         # The _ prefix in _name implies private ownership, but nothing is truly
         # private in Python.
 
     def test_you_can_also_access_the_value_out_using_getattr_and_dict(self):
         fido = self.Dog2()
         fido.set_name("Fido")
 
-        self.assertEqual(__, getattr(fido, "_name"))
+        self.assertEqual('Fido', getattr(fido, "_name"))
         # getattr(), setattr() and delattr() are a way of accessing attributes
         # by method rather than through assignment operators
```

Comment:
I think the main benefit is the fact that you can pass `getattr` a default value, so if for some reason the attribute it's not correctly taken, you can have other instead of going directly in a Exception.

Reply:
Yes, that is one very good reason. 

The other big advantage is that you can get the properties of things using strings. This gives you a lot of flexibility because you don't need to know the name of the parameter beforehand. As an example, here's a simple function that gets you the values of all properties you request.

```python
In [1]: def get_properties(object, properties): 
   ...:   """ 
   ...:   Gets the  
   ...:   :param object: Object from which to extract values. 
   ...:   :param properties: Properties to extract values for.
   ...:   """ 
   ...:   values = [] 
   ...:   for property in properties: 
   ...:     values.append(getattr(object, property, None)) 
   ...:  
   ...:   return values 
   ...:                                                                                                                                                                                                                                                                              

In [2]: class Test(object): 
   ...:   def __init__(self, a, b, c): 
   ...:     self.a = a 
   ...:     self.b = b 
   ...:     self.c = c 
   ...:                                                                                                                                                                                                                                                                              

In [3]: t = Test(1, 2, 3)                                                                                                                                                                                                                                                            

In [4]: get_properties(t, ('a', 'c', 'error'))                                                                                                                                                                                                                                                

In [5]: print(v)                                                                                                                                                                                                                                                                     
[1, 3, None]

```